index.js

```diff
@@ -105,6 +105,8 @@ function set(pathname, handler) {
         }
     }
 
+    hash.src = pathname;
+
     if (!hash.handler) {
         hash.handler = handler;
```

Comment:
You should really be defining hash.src inside the if statement, just in case someone believes they can recover from the exception...